src/main/java/Duke.java

```diff
@@ -1,10 +1,99 @@
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.ChronoUnit;
+
+/**
+ * Duke is a small educational project which acts
+ * based on user input.
+ * @author Xia Liyi
+ */
 public class Duke {
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+    	Scanner sc = new Scanner(System.in);
+        // String logo = " ____        _        \n"
+        //         + "|  _ \\ _   _| | _____ \n"
+        //         + "| | | | | | | |/ / _ \\\n"
+        //         + "| |_| | |_| |   <  __/\n"
+        //         + "|____/ \\__,_|_|\\_\\___|\n";
+        Ui ui = new Ui();
+        Parser parser = new Parser();
+        Storage file = new Storage("../data/duke.txt");
+        ui.showLine("Hello! I'm Duke\nWhat can I do for you");
+        TaskList taskList = new TaskList(file.getTasks());
+        while (sc.hasNext()) {
```

Comment:
Perhaps it might be cleaner to abstract this while loop into a separate class/method?